jetty-openid/src/main/java/org/eclipse/jetty/security/openid/OpenIdConfiguration.java

```diff
@@ -18,38 +18,40 @@
 
 package org.eclipse.jetty.security.openid;
 
-import java.io.InputStream;
 import java.io.Serializable;
-import java.net.URI;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
 import java.util.Map;
 import java.util.Objects;
+import java.util.stream.Collectors;
 
-import org.eclipse.jetty.util.IO;
+import org.eclipse.jetty.client.HttpClient;
 import org.eclipse.jetty.util.ajax.JSON;
+import org.eclipse.jetty.util.component.ContainerLifeCycle;
 import org.eclipse.jetty.util.log.Log;
 import org.eclipse.jetty.util.log.Logger;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
 
 /**
  * Holds the configuration for an OpenID Connect service.
  *
  * This uses the OpenID Provider URL with the path {@link #CONFIG_PATH} to discover
  * the required information about the OIDC service.
  */
-public class OpenIdConfiguration implements Serializable
+public class OpenIdConfiguration extends ContainerLifeCycle implements Serializable
```

Comment:
So good that you looked into this. I had no idea about these things.

Reply:
Not too happy with this yet, doesn't seem a great idea to have the configuration as a LifeCycle. It would be good to have the HttpClient managed externally however it makes the `OpenIdConfiguration(String provider, String clientId, String clientSecret)` constructor difficult.

Reply:
Wrap it in a `ManagedObject` class that is just an input to the config? Where and how to create that `ManagedObject` object though is the question. That's where I was trying to go with the factory I had.

Reply:
If it is an active object (ie has a lifecycle) then perhaps it is not a "Configuration", rather a "Resources".  Also is it really Serializable now it is a lifecycle?  I think not.

Reply:
Can we change the name of `OpenIdConfiguration` wouldn't that be a breaking change?